### DISCUSSION TITLE
Short market orders

### DISCUSSION THREAD

**Discussion by petmo** [gbeced/basana#40] [General]
Hi,

love the package. 

See there is a working branch for supporting loans / margin trading in backtest exchange. However, is there currently a way to make short trades via the binance exchange?

**Comment by gbeced** [comment]
Hi. Thanks for your kind comment!

Yes, Basana lets you use Binance cross margin and isolated margin accounts.

Here are links to relevant functionality:

1. Use one of these attributes to get right margin account (note that the examples use the spot account):
 * https://basana.readthedocs.io/en/latest/binance_exchange.html#basana.external.binance.exchange.Exchange.cross_margin_account
 * https://basana.readthedocs.io/en/latest/binance_exchange.html#basana.external.binance.exchange.Exchange.isolated_margin_account

2. Depending on the margin account that you're using, use one of these two methods to fund the account:
 * https://basana.readthedocs.io/en/latest/binance_cross.html#basana.external.binance.cross_margin.Account.transfer_from_spot_account
 * https://basana.readthedocs.io/en/latest/binance_isolated.html#basana.external.binance.isolated_margin.Account.transfer_from_spot_account
 
3. Create your orders setting *side_effect_type* to *MARGIN_BUY* for auto borrowing and to *AUTO_REPAY* for auto repaying.
